mlrun/feature_store/retrieval/spark_merger.py

```diff
@@ -185,31 +103,37 @@ def _asof_join(
         # set join conditions
         join_cond = (
             entity_with_id[entity_timestamp_column]
-            >= aliased_featureset_df[f"ft__{entity_timestamp_column}"]
+            >= aliased_featureset_df[
+                rename_right_keys.get(entity_timestamp_column, entity_timestamp_column)
+            ]
         )
 
         # join based on entities
-        for key in indexes:
+        for key_l, key_r in zip(left_keys, right_keys):
             join_cond = join_cond & (
-                entity_with_id[key] == aliased_featureset_df[f"ft__{key}"]
+                entity_with_id[key_l]
+                == aliased_featureset_df[rename_right_keys.get(key_r, key_r)]
             )
 
         conditional_join = entity_with_id.join(
             aliased_featureset_df, join_cond, "leftOuter"
         )
-        for key in indexes + [entity_timestamp_column]:
-            conditional_join = conditional_join.drop(
-                aliased_featureset_df[f"ft__{key}"]
-            )
+        for key in right_keys + [entity_timestamp_column]:
+            if f"ft__{key}" in conditional_join.columns:
+                conditional_join = conditional_join.drop(
+                    aliased_featureset_df[f"ft__{key}"]
+                )
 
-        window = Window.partitionBy("_row_nr", *indexes).orderBy(
+        window = Window.partitionBy("_row_nr", *left_keys).orderBy(
```

Comment:
Why do you need the `*left_keys` in the `partitionBy` clause? Since you're partitioning by the left-side (== `entity-df`) row number, every other `left` key will be equal within the same partition. Shouldn't this be `*right_keys`?